src/llama-kv-cells.h

```diff
@@ -0,0 +1,259 @@
+#pragma once
+
+#include <bitset>
+#include <cassert>
+#include <vector>
+
+using llama_pos    = int32_t;
+using llama_seq_id = int32_t;
+
+// meta information about KV cells that can be part of multiple sequences at the same time
+// TODO: add unit tests
+struct llama_kv_cells_unified {
```

Comment:
Since there are private members, I would recommend changing this to a class in accordance with [CppCoreGuidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#c8-use-class-rather-than-struct-if-any-member-is-non-public).